esmvaltool/interface_scripts/auxiliary.ncl

```diff
@@ -1028,3 +1088,259 @@ begin
   return(str)
 
 end
+
+; #############################################################################
+undef("get_average")
+function get_average(items: list)
+
+;
+; Arguments
+;    items: list of input_file_info metadata
+;
+; Description
+;    Calculates average of all datasets given by items.
+;
+; Caveats
+;
+; References
+;
+; Modification history
+;    20181217-schlund_manuel: written
+;
+local funcname, scriptname, dim_dat, dim_array, rank, data_array, data
+begin
+
+  funcname = "get_average"
+  scriptname = "interface_scripts/auxiliary.ncl"
+  enter_msg(scriptname, funcname)
+
+  ; Check input list
+  dim_dat = ListCount(items)
+  if (dim_dat .lt. 1) then
+    error_msg("f", scriptname, funcname, "expected at least one dataset, " + \
+              "got empty list")
+  end if
+
+  ; Collect data
+  do idat = 0, dim_dat - 1
+    data = read_data(items[idat])
+
+    ; Create array at first iteration
+    if (.not. isvar("data_array")) then
+      dim_array = dimsizes(data)
+      rank = dimsizes(dim_array)
+      data_array = new(array_append_record(dim_dat, dim_array, 0), \
+                       typeof(data))
+    end if
+    if (rank .eq. 4) then
+      copy_VarCoords(data, data_array(0, :, :, :, :))
+      data_array(idat, :, :, :, :) = (/data/)
+    else if (rank .eq. 3) then
```

Comment:
Using `elseif` would make this block shorter.

Reply:
What's the difference between `else if` and `elseif` in NCL?

Reply:
`else if` is the old version, which requires an `end if` for each of it.